tests/test_envrc.py

```diff
@@ -0,0 +1,52 @@
+from itertools import permutations
+import os
+
+# Define the cookiecutter directory
+DIR_COOKIECUTTER = "{{ cookiecutter.repo_name }}"
+
+# Initialise an empty list to store the expected filepaths
+DIRS_EXPECTED = []
+
+# Get all the directories within `DIR_COOKIECUTTER`, including all subfolders (unless they are `docs` subfolders)
+for root, dirs, files in os.walk(DIR_COOKIECUTTER):
+    if not os.path.join(DIR_COOKIECUTTER, "docs") in root:
+        DIRS_EXPECTED += list(map(lambda d: os.path.join(os.getcwd(), root, d), dirs))
+
+# Get the expected environment variable names for the directories in `DIRS_EXPECTED`; we expect directory environment
+# variables to start with "DIR_", and then the path from the top-level to the folder in uppercase, where "/" is
+# replaced with "_". For example, a folder called "foo" with a sub-folder "bar" should have environment variables
+# "DIR_FOO" and "DIR_FOO_BAR", respectively
+ENV_DIRS_EXPECTED = [
+    "DIR" + p.replace(os.path.join(os.getcwd(), DIR_COOKIECUTTER), "").replace("/", "_").upper() for p in DIRS_EXPECTED
+]
```

Comment:
```suggestion
ENV_DIRS_EXPECTED = [
    "DIR" + p.replace(os.path.join(os.getcwd(), DIR_COOKIECUTTER), "").replace("/", "_").upper() for p in DIRS_EXPECTED if not p.endswith("___pycache__")
]
```

To exclude "__pycache__" subdirectories.

Reply:
Shouldn't be needed if `DIRS_EXPECTED` is fixed for the `__pycache__` issue. Marking as **Resolved**.